src/main/java/org/osiam/auth/exception/OsiamExceptionHandler.java

```diff
@@ -60,6 +60,14 @@ protected JsonErrorResult handleResourceNotFound(HttpServletRequest request,
         return new JsonErrorResult(HttpStatus.NOT_FOUND.name(), e.getMessage());
     }
 
+    @ExceptionHandler(ClientAlreadyExistsException.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    @ResponseBody
+    protected JsonErrorResult handleClientAlreadyExists(ClientAlreadyExistsException e) {
+        LOGGER.warn("A ClientAlreadyExistsException occurred", e);
```

Comment:
Well, that the Exception occurred is easily visible from the delivered exception. I'm not even sure if I like the stack trace here, but I definitely would prefer a better log message, What do you think of "Tried to add client with id %id twice"?